docs/src/ref_impl/installer.md

```diff
@@ -5,67 +5,56 @@
-
-1. Set `ghaf.installer.enable` to `true`.
-2. Add nixos-generators module to `ghaf.installer.imgModules` list to configure installer image type.
-3. Choose installer modules from `ghaf.installer.installerModules` and set `ghaf.installer.enabledModules` to list of their names.
-4. Write code for the installer in `ghaf.installer.installerCode`.
-
-```nix
-{config, ...}: {
-  ghaf.installer = {
-    enable = true;
-    imgModules = [
-      nixos-generators.nixosModules.raw-efi
-    ];
-    enabledModules = ["flushImage"];
-    installerCode = ''
-      echo "Starting flushing..."
-      if sudo dd if=${config.system.build.${config.formatAttr}} of=/dev/${config.ghaf.installer.installerModules.flushImage.providedVariables.deviceName} conv=sync bs=4K status=progress; then
-          sync
-          echo "Flushing finished successfully!"
-          echo "Now you can detach installation device and reboot to Ghaf."
-      else
-          echo "Some error occured during flushing process, exit code: $?."
-          exit
-      fi
-    '';
-  };
-}
-```
+Ghaf has a NixOS-like non-interactive, declarative installer instead of a
+conventional (imperative, non-declarative) installer as in most conventional
+Linux distributions. This is possible with the [Ghaf as
+Library](https://github.com/tiiuae/ghaf/pull/ghaf-based-project.md) approach:
+rather than clicking similar options during installation, you can configure the
+system once and deploy this configuration to the desired machines.
+
+
+To implement the pre-configured setting up approach, we used the
+[nixos-anywhere](https://github.com/nix-community/nixos-anywhere) tool.
+
+There are two separate ways to use it:
+* manually
+* with the Ghaf installation wizard.
+> NOTE: The Ghaf installation wizard is currently under development and cannot be used to create the required system configuration.
 
-After that you can build an installer image using this command:
+## Manual Installation
+
+To install Ghaf manually:
+
+1. Create your own flake using the Ghaf template:
 
 ```sh
-nix build .#nixosConfigurations.<CONFIGURATION>.config.system.build.installer
+nix flake init -t github:tiiuae/ghaf#target-x86_64-generic
 ```
 
-## Adding Installer Modules
+2. Edit it according to your preferences.
+
+3. Set the value of `ghaf.installer.sshKeys` to get an installer image.
 
-To add an installer module, replace the corresponding placeholders with your code and add this to your configuraiton:
+4. Build the installer image:
 
-```nix
-ghaf.installer.installerModules.<MODULE_NAME> = {
-  requestCode = ''
-    # Your request code written in Bash
-  '';
-  providedVariables = {
-    # Notice the dollar sign before the actual variable name in Bash.
-    <VARIABLE_NAME> = "$<VARIABLE_NAME>";
-  };
-};
+```sh
+nix build .#nixosConfigurations.PROJ_NAME-ghaf-debug.config.system.build.installer
 ```
 
-## Built-in Installer Modules
+5. Flash the installer image to your device:
```

Comment:
Actually no - the `installer image` here refers to image that you flash to `your device` (which means either USB SSD or USB memory). 

Thanks for raising that this part of the documentation is ambiguous. Especially without examples.

Reply:
So is it then defined somewhere what is running on the target device? Some linux with root access?

Reply:
Yes, the installer image that you build is a modified NixOS installer image. You’ll notice when you boot the installer image in the target device. Please note that you don’t use it to install. It only enables ghaf remote install securely after you have connected the X1 to network.

Reply:
So NixOS linux with root access.

Reply:
Ok, so the idea is that the installer image (.iso) is flashed on USB device and then the target device is booted from that. Then connect to target device to network check ip and run the install command from the installer machine. This is what I have been doing but for a moment I though that "your device" meant the installer machine.

Is it required that also target device has access to internet or can it be in separate local network and only installer machine having internet access?

Reply:
> This is what I have been doing but for a moment I though that "your device" meant the installer machine.

I agree that "your device" is ambiguous in this context. Needs improvement - review and request improvements.

> Is it required that also target device has access to internet or can it be in separate local network and only installer machine having internet access?

Yes, I think so. I tested it so that the target device was connected to the internet. This is also good addition to doc as requirement.